teps/0053-nested-triggers.md

```diff
@@ -201,4 +202,86 @@ spec:
 Notice how simple it is for the Trigger author to write the logic around their trigger
 when the common logic is implemented at the top level. Trigger authors can also manage
 their own set of downstream triggers as you can specify additional triggers alongside
-a binding and template once the interceptor chain completes.
\ No newline at end of file
+a binding and template once the interceptor chain completes.
+
+## Alternatives
+
+There are three alternatives that have been identified to provide some of the same benefits.
+
+### Option 1: Select Downstream Triggers With Labels
+
+Add a `triggerSelector` field that allows a trigger to target downstream triggers based off a label selector:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: Trigger
+metadata:
+  name: top-level-trigger
+spec:
+  interceptors:
+    - cel:
+        filter: >-
+          header.match('header-filter', 'a-header-filter')
+        - key: example-overlay
+          expression: body.example.value
+  triggerSelector:
+    matchLabels:
+      key: value
+```
+
+### Option 2: Interceptor Groups
+
+Specify the interceptor groups at the top level and process each group on trigger:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: my-el
+spec:
+  groups: 
+  - name: github-group
+    interceptors:
+      - my-interceptor
+      - add-extra-body
+    triggerSelector:
+      matchLabels:
+        app=foo
+        anotherval=xyz
+  - name: another-group
+    interceptors:
+      - my-interceptor
+    triggerSelector:
+       matchLabels:
+         app=foo
+```
+
+This can work well with running a root EventListener interceptor with a `labelSelector`. One challenge here is that these `InterceptorGroups` become another separate API resource that would need to be managed and potentially added to the API. This could be added onto the existing Trigger resource, but then it becomes closer to Option 1.
+
+### Option 3: Downstream EventListener
```

Comment:
@wlynch  you had discussed this on a Triggers WG a few weeks back, but i don't know if I did the API spec justice here. Let me know if you had a different way to structure this downstream event listener target within a Trigger

Reply:
Sorry about missing this!

I think was I was thinking about was something more in line with Approach #2, but with additional config to say "invoke these triggers OR these EventListeners"

The primary thing I was interested in was having a clear separation between config and executable units. 
We might need to better define the boundaries between eventlisteners and triggers to reduce unexpected behavior.

Triggers today are a config object, not an executable object - a single trigger config can be referenced by  multiple event listeners. This is somewhat confusing because you can't actually call a trigger - you need to call the eventlistener that references the trigger config (perhaps with extra path/param information so we can target the particular config). This is the main motivation for referencing ELs instead of Triggers comes from.

Some of the things I'm worried about are:

- What identity do sub-triggers run as? Does it use the same identity as the current event listener, or a separate one?
- What is the behavior if multiple event listeners reference the same trigger config? Do we need to have additional selector config to restrict this?
- Are sub-triggers executed with the same event id as its parent? This has an impact on tracing, since we'd either have multiple triggers reporting to the same event id, or need to have a mechanism to associate multiple ids together.

This might be a sign that we need to refine how we model Triggers w.r.t. their relation to EventListeners, and make them more of an executable unit (though this might have it's own reusability drawbacks).